Spring-server/src/main/java/ikpi63holding/postclient/jwt/filters/JwtAuthorizationFilter.java

```diff
@@ -11,18 +13,19 @@
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
 
 public class JwtAuthorizationFilter extends BasicAuthenticationFilter {
 
     private final JwtProperties jwtProperties;
-    private final UserRepository userRepository;
+    private final UserPrincipalDetailsService principalDetailsService;
```

Comment:
I think using UserPrincipalDetailsService here makes more sense and we can forget about userRepository on security configuration level